commands/commands.go

```diff
@@ -97,6 +97,75 @@ func init() {
 		help: "Gives an inspirational quote",
 		exec: inspireCommand,
 	}
+
+	commMap["minecraft"] = &command{
+		help: "Check to see who is currently online on the NetsocCraft server",
+		exec: minecraftCommand,
+	}
+}
+
+// minecraftCommand checks the stats of minecraft.netsoc.co for that moment
+// data comes from http://minecraft.netsoc.co/standalone/dynmap_NetsocCraft.json
+
+func minecraftCommand(ctx context.Context, s *discordgo.Session, m *discordgo.MessageCreate, _ []string) error {
+	l, loggerOk := logging.FromContext(ctx)
+	if loggerOk {
+		l.Infof("Responding to minecraft command", nil)
+	}
+
+	resp, err := http.Get("http://minecraft.netsoc.co/standalone/dynmap_NetsocCraft.json")
+
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve data from the Minecraft Server: %v", err)
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("Failed to read the response body: %v", err)
+	}
+
+	// Unmarshal the data and get all the player lists I guess
+	q := &struct {
```

Comment:
I know that we're going to be building on this command so this is only an FYI more than a change request :P 

The only field you use out of the struct is the q.Players.Name field. So iirc, this could be shortened to:
```golang
q := &struct{
     Players []struct {
        Name string `json:"name"`
    } `json:"players"`
}{}
```

Reply:
I'm completely aware of this, I did take the extra stuff out and it ran perfectly, but I decided to leave it in for the purposes of possible extension :)

Reply:
I was thinking as much :)